framework/src/auxkernels/BuildArrayVariableAux.C

```diff
@@ -0,0 +1,58 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BuildArrayVariableAux.h"
+
+#include <algorithm>
+
+registerMooseObject("MooseApp", BuildArrayVariableAux);
+
+InputParameters
+BuildArrayVariableAux::validParams()
+{
+  InputParameters params = ArrayAuxKernel::validParams();
+  params.addCoupledVar("component_variables",
+                       "The variables that make up each component of the output array variable.");
+  params.addClassDescription("Combines multiple standard variables into an array variable.");
+  return params;
+}
+
+BuildArrayVariableAux::BuildArrayVariableAux(const InputParameters & parameters)
+  : ArrayAuxKernel(parameters), _component_dofs(coupledAllDofValues("component_variables"))
+{
+  // Check the number of component variables
+  if (_component_dofs.size() != _var.count())
+    paramError("variable",
+               "The array variable has ",
+               _var.count(),
+               " components, but ",
+               _component_dofs.size(),
+               " component variables were specified.");
+
+  // Make sure the FEType of each input variable matches the output type
+  for (const auto & var : getCoupledMooseVars())
+    if (var->feType() != _var.feType())
+      paramError(
+          "component_variables",
+          "The input and output variables of BuildArrayVariableAux must have the same FE type");
```

Comment:
```suggestion
          "The input and output variables must have the same FE type");
```

I had this one in a previous comment, I think it was missed. Extraneous info because the param error reports this